### PR TITLE
Toolbar: buttons declared via template should be hidden if there is no enough free space for them (T1191856)

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

**Review by iBat — DISMISSED** [review]

**Review by alexanderPolosatov — DISMISSED** [review]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/toolbar.tests.js [hunk 1]
```diff
@@ -930,6 +931,76 @@ QUnit.module('adaptivity', moduleConfig, () => {
         assert.strictEqual($visibleItems.length, 3, 'two items was hidden');
     });
 
+    QUnit.test('Buttons declared via template should be hidden if there is no enough free space for them (T1191856)', function(assert) {
+        this.instance.option({
+            items: [
+                {
```
Comment (AlexanderMoiseev) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/toolbar.tests.js]: [minor] i'd use const to get rid of duplication
```

const itemOptions = getOptions(location, widget, locateInMenu) =>  {
                    location,
                    widget,
                    locateInMenu,
                    template() {
                        return $('<div />').dxButton({
                            text: 'Button long text'
                        });
                    }
                }

```

or, as is see params for all items expect the last one are equal, so this'll do
```

 const itemOptions = {
                    location: 'before',
                    widget: 'dxButton',
                    locateInMenu: 'auto',
                    template() {
                        return $('<div />').dxButton({
                            template: 'Button long text'
                        });
                    }
                },
```

**Review by AlexanderMoiseev — DISMISSED** [review]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/toolbar.tests.js [hunk 2]
```diff
@@ -930,6 +931,76 @@ QUnit.module('adaptivity', moduleConfig, () => {
...
+                    location: 'before',
+                    widget: 'dxButton',
+                    locateInMenu: 'auto',
+                    template(e, f, g) {
```
Comment (AlexanderMoiseev) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/toolbar.tests.js]: seems like these params are unused

Comment (EugeniyKiyashko) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/toolbar.tests.js]: https://github.com/DevExpress/DevExtreme/pull/25740/commits/92018c3279c27a839aaf199c2a638b4b487b8861

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/toolbar.tests.js (hunk 1 above)
Comment (EugeniyKiyashko) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/toolbar.tests.js]: https://github.com/DevExpress/DevExtreme/pull/25740/commits/92018c3279c27a839aaf199c2a638b4b487b8861

**Review by alexanderPolosatov — APPROVED** [review]

**Review by AlexanderMoiseev — APPROVED** [review]

**Pull request merged**